onmt/data/SampledDataset.lua

```diff
@@ -11,8 +11,13 @@ local options = {
     }
   },
   {
-    '-sample_w_ppl', false,
-    [[If set, ese perplexity as a probability distribution when sampling.]]
+    '-sample_type', 'uniform',
+    [[Define the partition type. `uniform` draws randomly the sample, `w_ppl` uses perplexity
+      as a probability distribution when sampling (with `-sample_w_ppl_init` and `-sample_w_ppl_max`
+      options), `partition` draws different subsets at each epoch.]],
+    {
+      enum = { 'uniform', 'w_ppl', 'partition'}
```

Comment:
Just for the style, I would also take the chance to replace `w_ppl` by `perplexity`.

```
-sample perplexity
-sample_perplexity_init
-sample_perplexity_max
```

(And also regenerate the options documentation.)